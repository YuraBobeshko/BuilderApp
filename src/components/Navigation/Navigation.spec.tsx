import 'jsdom-global/register';
import {Navigation} from "../index";
import {shallow} from "enzyme";
import { useParams } from "react-router-dom";

const currentId = { currentId: '1111' }

let mockCallback = jest.fn(() => currentId)

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: jest.fn(() => currentId),
}))

const setUp = () => shallow(<Navigation />);


describe('Navigation component', () => {

    it('should render Navigation component with currentId', () => {
        let component = setUp();
        expect(component).toMatchSnapshot();
    })

    it('should render Navigation component without currentId', () => {
        // @ts-ignore
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        useParams = () => ({
            currentId: null
        })
        let component = setUp();

        expect(component).toMatchSnapshot();
    })

})
