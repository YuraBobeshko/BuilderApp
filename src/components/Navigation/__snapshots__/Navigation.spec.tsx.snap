// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render Navigation component with currentId 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <li
    key="ListProject"
  >
    <Link
      to="/ListProject/1111"
    >
      ListProject
    </Link>
  </li>
  <li
    key="Structure"
  >
    <Link
      to="/Structure/1111"
    >
      Structure
    </Link>
  </li>
  <li
    key="Builder"
  >
    <Link
      to="/Builder/1111"
    >
      Builder
    </Link>
  </li>
  <li
    key="Config"
  >
    <Link
      to="/Config/1111"
    >
      Config
    </Link>
  </li>
</div>
`;

exports[`Navigation component should render Navigation component without currentId 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
/>
`;
